android/app/src/main/java/com/tgmu/tgmu/ui/fragment/ExperiencesFragment.kt

```diff
@@ -48,15 +46,28 @@ class ExperiencesFragment : Fragment(R.layout.fragment_experiences) {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        binding.btnAddExperience.setOnClickListener {
+            experienceViewModel.setDefaultExperience()
+            val action =
+                ExperiencesFragmentDirections.experienceViewToExperienceForm()
+            findNavController().navigate(action)
+        }
+
         val searchAdapter = MovieSearchSuggestionsAdapter {
             binding.sbMovie.setText(it.title)
             binding.svMovie.hide()
             experienceViewModel.getExperiencesByMovieId(it.id)
         }
 
-        val experienceAdapter = CompactExperienceAdapter() {
+        val experienceAdapter = CompactExperienceAdapter(onLikeClicked = {
             experienceViewModel.toggleLiked(it, Firebase.auth.currentUser!!.uid)
-        }
+        }, onEditClicked = {
+            experienceViewModel.setSpecificExperience(it)
+            val action =
+                ExperiencesFragmentDirections.experienceViewToExperienceForm(it)
```

Comment:
Do you think it would maybe be better to pass the experience as an argument to the next fragment? This way if the experience is null it means a new experience and if the experience is not null it means an edit it required. In the creation of the form fragment it would be possible to set the experienceFormData and then the setSpecificExperience logic will be removed from this fragment.